openassessment/management/tests/test_create_oa_submissions_from_file.py

```diff
@@ -0,0 +1,357 @@
+                self.cmd.read_config_file(nonexistant_path)
+
+    def test_read_config_file__error(self):
+        """ Test for error behavior when test file cannot be read successfuly """
+        invalid_json = '{ this ] aint ( valid " json :( /'
+        with self._test_config_file(invalid_json) as (_, config_path):
+            with self.assertRaisesRegex(CommandError, f'Unable to parse file {config_path}'):
+                self.cmd.read_config_file(config_path)
+
+    def test_load_ora_blocks__multiple_blocks(self):
+        """
+        Test for error behavior when we attempt to look up a block
+        in a course that multiple blocks with that display name
+        """
+        mock_blocks = [
+            Mock(display_name='ORA A', location=1),
+            Mock(display_name='ORA B', location=2),
+            Mock(display_name='ORA A', location=3),
+        ]
+        expected_error_message = (
+            "The ORA 'ORA A' is specified in the input file. "
+            "The course contains more than one ORA with that display name. "
+            "First two found:  [1, 3]"
+        )
+        with patch.object(self.cmd, '_load_ora_blocks_from_modulestore') as mock_modulestore:
+            mock_modulestore.return_value = mock_blocks
+            with self.assertRaisesMessage(CommandError, expected_error_message):
+                self.cmd.load_ora_blocks(COURSE_ID, ['ORA A', 'ORA B'])
+
+    def test_load_ora_blocks__missing(self):
+        """
+        Test for error behavior when we are unable to find a block of
+        a given display name
+        """
+        mock_blocks = [
+            Mock(display_name='ORA A', location=1),
+            Mock(display_name='ORA B', location=2),
+            Mock(display_name='ORA C', location=3),
+        ]
+        expected_error_message = f'The following Display Name(s) were not found in {COURSE_ID}'
+        with patch.object(self.cmd, '_load_ora_blocks_from_modulestore') as mock_modulestore:
+            mock_modulestore.return_value = mock_blocks
+            with self.assertRaisesMessage(CommandError, expected_error_message) as err:
+                self.cmd.load_ora_blocks(COURSE_ID, ['ORA A', 'ORA B', 'ORA F', 'ORA Z'])
+            error_message = str(err.exception)
+            assert 'ORA F' in error_message
+            assert 'ORA Z' in error_message
+
+    def test_load_anonymous_user_ids__missing(self):
+        """ Test for when we are unable to find an anonymous_user_id for a username """
+
+        def mock_anonymous_id_for_user(user, course_id):  # pylint: disable=unused-argument
+            return user.username + "_anonymous"
+
+        for username in [USERNAME_1, USERNAME_2, STAFF_USER_1, STAFF_USER_2, SUPERUSER_USERNAME]:
+            UserFactory.create(username=username)
+
+        expected_error = 'Unable to load anonymous id for user(s) user_not_found'
+        with patch.object(self.cmd, '_import_anonymous_id_for_user', return_value=mock_anonymous_id_for_user):
```

Comment:
Don't like patching class internals. I'm actually surprised that pylint isn't yelling at you for this, it usually does.

Reply:
mm, I don't necessarily agree with that, but I can see where you're coming from